### PR TITLE
feat: add vector with defaults

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR adds Vector as a dependency and sets defaults for:
- Extracting application logs (lms, cms, ingress, cert-manager, etc)
- Extracting tracking logs from Open edX.
- Storing compressed tracking logs in `gzip` files on an S3-like storage (can be tweaked to work with Cloudwatch, Minio, etc)  by date.

Closes: #32

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/openedx-k8s-harmony-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: charts/harmony-chart/values.yaml
Comment (bmtcril) [inline charts/harmony-chart/values.yaml:445]: How is this going to work with the existing Aspects vector settings? Are the two going to be exclusive?

Comment (Ian2012) [reply charts/harmony-chart/values.yaml:445]: Yes, both need to be exclusive, for resource consumption and pipeline tracking. However, another filter and a sink can be added to filter specifics of any namespace.

However, keep in mind that Ralph is the recommended pipeline to use.

I can leave examples for it

File: charts/harmony-chart/values.yaml
Comment (gabor-boros) [inline charts/harmony-chart/values.yaml:465]: I would put this example (and the one in L533) into the https://github.com/openedx/openedx-k8s-harmony/blob/main/values-example.yaml. From helm chart point of view, this looks good to me, though I didn't check the setup.

**Review by gabor-boros — APPROVED** [review]

**Comment by Ian2012** [comment]
I've updated the default values as the sinks cannot be disabled, they need to be removed. Examples were added to setup CloudWatch and S3 sinks. This was tested on a production environment with CloudWatch, S3, and VictoriaLogs sinks, everything worked well and logs for the `lms|cms`  are grouped when there are exceptions.

File: charts/harmony-chart/values.yaml
Comment (Ian2012) [reply charts/harmony-chart/values.yaml:467]: I've updated the values-example.yaml with the elements for one installation filtered by namespace. It just needs to be replicated for every openedx namespace with Aspects installed.

File: values-example.yaml
Comment (MoisesGSalas) [inline values-example.yaml:146]: This are the tutor jinja variables right? I think in the context of a values file it might be a bit confusing considering the helm templates also use the brace syntax. Maybe you can use some example values and include in the comment each variable, like:

```
# Example ClickHouse Sink
# ASPECTS_CLICKHOUSE_VECTOR_USER: ch_vector_user
# ASPECTS_CLICKHOUSE_VECTOR_PASSWORD: hunter2
```

**Comment by jfavellar90** [comment]
@Ian2012 @gabor-boros I rebased the branch, bumped the Harmony version, and updated the `Chart.lock` file. LGTM now

**Review by jfavellar90 — APPROVED** [review]

**Pull request merged**
